crates/hyperqueue/src/dashboard/ui/screens/home/chart_section.rs

```diff
@@ -0,0 +1,147 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+use tako::messages::gateway::CollectedOverview;
+
+use crate::dashboard::ui::terminal::DashboardFrame;
+use tui::layout::Rect;
+use tui::style::{Color, Modifier, Style};
+use tui::symbols;
+use tui::text::Span;
+use tui::widgets::{Axis, Block, Borders, Chart, Dataset};
+
+/// A new WorkerTimeState is created everytime there is a #worker change event
+#[derive(Clone)]
+struct WorkerStateAtTime {
+    /// The time at which the event being stored happened, wrt. the reference
+    pub at_time: u128,
+    pub event: TimeEventType,
+}
+
+#[derive(Clone)]
+pub enum TimeEventType {
+    /// Current number of workers.
+    WorkerCountUpdate(u32),
+    /// The average cpu utilization of the worker.
+    AvgCpuUsageUpdate(f32),
+}
+
+#[derive(Clone)]
+pub struct ClusterOverviewChart {
+    /// The state of the worker at different times.
+    timed_info_vec: Vec<WorkerStateAtTime>,
```

Comment:
The data for the chart is updated in discrete steps, and in each step we get new information for each observed value on the Y axis. So I think that it would make more sense to represent the dataset as a collection of items (x_value, y_values). There's no need to repeat the X-value inside the Vec here.